openapi.yaml

```diff
@@ -132,120 +586,313 @@ securityDefinitions:
+        type: array
+        items:
+          $ref: '#/definitions/PizzaResource'
+      links:
+        $ref: '#/definitions/SelfLink'
+  PizzaResult:
+    properties:
+      data:
+        $ref: '#/definitions/PizzaResource'
+      links:
+        $ref: '#/definitions/SelfLink'
+  PizzaResource:
+    properties:
+      type: &pizzaType
+        type: string
+        description: the type of the resource (pizza)
+        enum: [pizza]
+      id: &pizzaId
+        type: string
+        description: a unique ID of a pizza
+        example: abc123
+      links:
+        $ref: '#/definitions/SelfLink'
+      attributes:
+        $ref: '#/definitions/PizzaAttributes'
+      relationships: &pizzaRelationships
+        properties:
+          dough:
+            required: [data]
+            properties:
+              links:
+                properties:
+                  self:
+                    type: string
+                    description: a link to the relationship
+                    example: https://api.oregonstate.edu/v1/pizzas/1/relationships/dough
+                  related:
+                    type: string
+                    description: a link to the dough referenced
+                    example: https://api.oregonstate.edu/v1/pizzas/1/dough
+              data:
+                $ref: '#/definitions/DoughRelationshipResource'
+          ingredients:
+            required: [data]
+            properties:
+              links:
+                properties:
+                  self:
+                    type: string
+                    description: a link to the relationship
+                    example: https://api.oregonstate.edu/v1/pizzas/1/relationships/dough
+                  related:
+                    type: string
+                    description: a link to the ingredients for the dough referenced
+                    example: https://api.oregonstate.edu/v1/pizzas/1/ingredients
+              data:
+                type: array
+                description: a list of relationship resources with the ingredient type
+                items:
+                  $ref: '#/definitions/IngredientRelationshipResource'
+      included: &pizzaIncluded
+        type: array
+        items:
+          type: object
+          description: An ingredient resource or a dough resource
+  PizzaPostBody:
+    required: [data]
+    properties:
+      data:
+        required: [type, attributes]
+        properties:
+          type: *pizzaType
+          attributes:
+            $ref: '#/definitions/PizzaPostAttributes'
+          relationships: &pizzaUpdateRelationships
+            properties:
+              dough:
+                properties:
+                  data:
+                    $ref: '#/definitions/DoughRelationshipResource'
+              ingredients:
+                properties:
+                  data:
+                    type: array
+                    items:
+                      $ref: '#/definitions/IngredientRelationshipResource'
```

Comment:
`properties` and `ingredients` definitely make sense to make optional, but I think some of the fields within these objects should be required. Could you take a look at this? I think this will need to be fixed for the other endpoints as well.

Reply:
Adding the `type` and `id` fields as required on each relationship resource object